exchanges/okcoin/okcoin.go

```diff
@@ -12,613 +12,1658 @@ import (
+	if withdrawalID == "" {
+		return "", errWithdrawalIDMissing
+	}
+	var resp []struct {
+		WithdrawalID string `json:"wdId"`
 	}
-	var resp []interface{}
-	err = o.SendHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, tokenSubsection, requestURL, nil, &resp, false)
+	err := o.SendHTTPRequest(ctx, exchange.RestSpot, fiatCancelWithdrawalEPL, http.MethodPost, typeFiat, "cancel-withdrawal", map[string]string{"wdId": withdrawalID}, &resp, true)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
-	candles := make([]kline.Candle, len(resp))
-	for x := range resp {
-		t, ok := resp[x].([]interface{})
-		if !ok {
-			return nil, common.GetTypeAssertError("[]interface{}", resp[x])
-		}
-		if len(t) < 6 {
-			return nil, fmt.Errorf("%w expteced %v received %v", errIncorrectCandleDataLength, 6, len(t))
-		}
-		v, ok := t[0].(string)
-		if !ok {
-			return nil, common.GetTypeAssertError("string", t[0])
-		}
-		if candles[x].Time, err = time.Parse(time.RFC3339, v); err != nil {
-			return nil, err
-		}
-		if candles[x].Open, err = convert.FloatFromString(t[1]); err != nil {
-			return nil, err
-		}
-		if candles[x].High, err = convert.FloatFromString(t[2]); err != nil {
-			return nil, err
-		}
-		if candles[x].Low, err = convert.FloatFromString(t[3]); err != nil {
-			return nil, err
-		}
-		if candles[x].Close, err = convert.FloatFromString(t[4]); err != nil {
-			return nil, err
-		}
-		if candles[x].Volume, err = convert.FloatFromString(t[5]); err != nil {
-			return nil, err
-		}
+	if len(resp) == 0 {
+		return "", errNoValidResponseFromServer
 	}
-	return candles, nil
+	return resp[0].WithdrawalID, nil
 }
 
-// GetMarginTradingAccounts List all assets under token margin trading account, including information such as balance, amount on hold and more.
-func (o *OKCoin) GetMarginTradingAccounts(ctx context.Context) ([]GetMarginAccountsResponse, error) {
-	var resp []GetMarginAccountsResponse
-	return resp, o.SendHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, marginTradingSubsection, accounts, nil, &resp, true)
+// GetFiatWithdrawalHistory retrieves a fiat withdrawal orders list
+// Channel ID used in the transaction.  9:PrimeX; 28:PrimeX US; 21:PrimeX Europe; 3:Silvergate SEN; 27:Silvergate SEN HK
+// Withdrawal state. -2:User canceled the order；-1:Withdrawal attempt has failed; 0:Withdrawal request submitted; 1:Withdrawal request is pending; 2:Withdrawal has been credited
+func (o *Okcoin) GetFiatWithdrawalHistory(ctx context.Context, ccy currency.Code, channelID, withdrawalState, withdrawalID string, after, before time.Time, limit int64) ([]FiatWithdrawalHistoryItem, error) {
+	params := url.Values{}
+	if channelID != "" {
+		params.Set("chanId", channelID)
+	}
+	if !ccy.IsEmpty() {
+		params.Set("ccy", ccy.String())
+	}
+	if withdrawalState != "" {
+		params.Set("state", withdrawalState)
+	}
+	if withdrawalID != "" {
+		params.Set("depId", withdrawalID)
```

Comment:
Should be `wdId`:
https://www.okcoin.com/docs-v5/en/#rest-api-fiat-get-withdrawal-history